packages/core/test/object_factory_tests.js

```diff
@@ -198,32 +203,79 @@ tap.test(
       '00-atoms/00-global/00-colors-alt/colors-alt.mustache',
       { d: 123 }
     );
-    test.equals(p.getDirLevel(0, { hasDir: true, dirLevel: 3 }), '00-atoms');
-    test.equals(p.getDirLevel(1, { hasDir: true, dirLevel: 3 }), '00-global');
-    test.equals(p.getDirLevel(3, { hasDir: true, dirLevel: 3 }), '');
+    test.equals(p.getDirLevel(0, { patternHasOwnDir: true }), '00-atoms');
+    test.equals(p.getDirLevel(1, { patternHasOwnDir: true }), '00-global');
+    test.equals(p.getDirLevel(2, { patternHasOwnDir: true }), ''); // There is no third level
+
     var p = new Pattern('00-atoms/00-colors-alt/colors-alt.mustache', {
       d: 123,
     });
-    test.equals(p.getDirLevel(0, { hasDir: true, dirLevel: 2 }), '00-atoms');
-    test.equals(
-      p.getDirLevel(1, { hasDir: true, dirLevel: 2 }),
-      '00-colors-alt'
-    );
-    test.equals(p.getDirLevel(3, { hasDir: true, dirLevel: 2 }), '');
+    test.equals(p.getDirLevel(0, { patternHasOwnDir: true }), '00-atoms');
+    test.equals(p.getDirLevel(1, { patternHasOwnDir: true }), ''); // There is no second level
+    test.equals(p.getDirLevel(2, { patternHasOwnDir: true }), ''); // There is no third level
+
     var p = new Pattern('00-colors-alt/colors-alt.mustache', { d: 123 });
-    test.equals(
-      p.getDirLevel(0, { hasDir: true, dirLevel: 1 }),
-      '00-colors-alt'
-    );
-    test.equals(
-      p.getDirLevel(1, { hasDir: true, dirLevel: 1 }),
-      '00-colors-alt'
+    test.equals(p.getDirLevel(0, { patternHasOwnDir: true }), 'root'); // No first level means root
+    test.equals(p.getDirLevel(1, { patternHasOwnDir: true }), ''); // There is no second level
+    test.equals(p.getDirLevel(2, { patternHasOwnDir: true }), ''); // There is no third leveL
+
+    var p = new Pattern(
+      '00-atoms/00-global/00-colors-alt/colors-alt~variant.mustache',
+      { d: 123 }
     );
-    test.equals(p.getDirLevel(3, { hasDir: true, dirLevel: 1 }), '');
+    test.equals(p.name, '00-atoms-00-global-00-colors-alt-variant');
+    test.equals(p.flatPatternPath, '00-atoms-00-global');
+    test.equals(p.patternBaseName, 'colors-alt-variant');
+
     test.end();
   }
 );
 
+tap.test('test Patterns that are nested deeper without own directory', function(
```

Comment:
I added additional test cases here just to be sure if changes are made everything is working correctly after that.